Cethien/fixes

cethien opened the pull request.

cethien committed:
fix: module name

- module name should be same as link to package. `go get github.com/jlucasnsilva/gomponents/alpine` results in error

cethien committed:
perf: align to standards

cethien committed:
docs: readme

cethien committed:
perf: moved to root

cethien reviewed (commented):
heyo. i wanted to use the lib, but it had some problems.
fixed up some stuff, so it should be usable now

jlucasnsilva commented on alpine_test.go:
```diff
@@ -105,11 +105,11 @@ func TestAlpine(t *testing.T) {
 
 	assert.Panics(t, func() {
-		attribute("foo", "bar", "baz")
+		g.Attr("foo", "bar", "baz")
```
I tried using `g.Attr(...)`. The problem is that it scapes some characters and Alpine expects valid JS expressions. That's why I wrote the `attribute` function.

Did you run the tests with `g.Attr(...)` again? Maybe I missed something.

cethien replied:
just discovered it myself and did some testing:
`g.Attr` always escapes, but the browser engines convert them just fine, so no gremlins there.

cethien replied:
just looked on the [tests of the htmx lib](https://github.com/maragudk/gomponents-htmx/blob/main/htmx_test.go), author is testing `g.Node` against `g.Node` instead of the final output. makes more sense tbh

cethien replied:
rewrote the tests, also added a Makefile to make test easier and create a cover file

jlucasnsilva replied:
Thank you very much for your work. I've merged your pull request.

jlucasnsilva commented:
> heyo. i wanted to use the lib, but it had some problems. fixed up some stuff, so it should be usable now

Thank you very much. I wasn't expecting users so soon.

cethien commented:
> > heyo. i wanted to use the lib, but it had some problems. fixed up some stuff, so it should be usable now
> 
> Thank you very much. I wasn't expecting users so soon.

i have a project which was using [templ](https://github.com/a-h/templ) with alpine, htmx and hyperscript, but it doesnt utilize comment tags [yet](https://github.com/a-h/templ/issues/625) so I moved to gomponents

figured maybe others may try too, but due to your module name the lib is unusable right now, so here my pr 😄

cethien committed:
test: rewrite

- now tests nodes instead of string, to comform with gomponents

cethien committed:
perf: Makefile, format, gitignore
